### PR TITLE
sessionctx, executor: add session var to control explicit insertion on auto_random column

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Problem Summary: 

As the documentation says, 
> It is __NOT__ recommended that you __explicitly specify__ a value for the column with the AUTO_RANDOM attribute when you insert data. Otherwise, the numeral values that can be automatically assigned for this table might be used up in advance.

This PR adds a session variable named `tidb_allow_auto_random_explicit_insert`, which controls whether explicit insertion on auto_random column is allowed. By default, it is `false`.

### What is changed and how it works?

What's Changed:
- Add a new session variable `tidb_allow_auto_random_explicit_insert`.
- Add a few tests. 

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Breaking backward compatibility (Fortunately, auto_random is under experimental.)

### Release note 
- a new session variable named `tidb_allow_auto_random_explicit_insert` is added, which controls whether explicit insertion on auto_random column is allowed.

### CODE REVIEW

**Comment by tangenta** [comment]
> Why session variable, not global variable?

@djshow832 Although I think the case is rare, imagine there are two sessions:
session 1: importing data from the migration tool to table `t1`
session 2: inserting data from users to table `t2`, which needs the protection provided by this variable.

Global variable cannot meet the demand.

File: meta/autoid/errors.go [hunk 1]
```diff
@@ -47,6 +47,8 @@ const (
 	AutoRandomNonPositive = "the value of auto_random should be positive"
 	// AutoRandomAvailableAllocTimesNote is reported when a table containing auto_random is created.
 	AutoRandomAvailableAllocTimesNote = "Available implicit allocation times: %d"
+	// AutoRandomExplicitInsertDisabledErrMsg is reported when auto_random column value is explicitly specified, but the session var 'allow_auto_random_explicit_insert' is false.
+	AutoRandomExplicitInsertDisabledErrMsg = "Explicit insertion on auto_random column is disabled by default. Try to set @@allow_auto_random_explicit_insert = true."
```
Comment (djshow832) [inline meta/autoid/errors.go:51]:
```suggestion
	AutoRandomExplicitInsertDisabledErrMsg = "Explicit insertion on auto_random column is disabled. Try to set @@allow_auto_random_explicit_insert = true."
```
He might have set it to `false` manually.

**Pull request merged**